yarn-project/foundation/src/json-rpc/client/safe_json_rpc_client.ts

```diff
@@ -31,24 +62,100 @@ export function createSafeJsonRpcClient<T extends object>(
 ): T {
   const fetch = config.fetch ?? defaultFetch;
   const log = config.log ?? createLogger('json-rpc:client');
-  const { useApiEndpoints = false, namespaceMethods = false } = config;
+  const { namespaceMethods = false, batchWindowMS = DEFAULT_BATCH_WINDOW_MS } = config;
 
   let id = 0;
+  let sendBatchTimeoutHandle: NodeJS.Timeout | undefined;
+  let queue: Array<{ request: JsonRpcRequest; deferred: PromiseWithResolvers<JsonRpcResponse> }> = [];
+
+  const sendBatch = async () => {
+    if (sendBatchTimeoutHandle !== undefined) {
+      clearTimeout(sendBatchTimeoutHandle);
+      sendBatchTimeoutHandle = undefined;
+    }
+
+    const rpcCalls = queue;
+    queue = [];
```

Comment:
Js question: I take it that the `request` closure below always reads the freshest value of `queue`, right? And not captures it?

Reply:
yes, the closure would capture the variable, not the value

Reply:
TIL!